testnet2/testnet2.toml

```diff
@@ -6,28 +6,28 @@ Key = "network"
 Value = "testnet2"
 
 [API]
-
   [API.GRPC]
     Hosts = [
       "api-n00.validators-testnet.vega.rocks:3007",
       "api-n02.validators-testnet.vega.rocks:3007",
       "api-n03.validators-testnet.vega.rocks:3007",
       "api-n04.validators-testnet.vega.rocks:3007",
       "api-n05.validators-testnet.vega.rocks:3007",
-      "vega-testnet.anyvalid.com:3007"
+      "vega-testnet.anyvalid.com:3007",
+      "testnet.grpc.vega.xprv.io:443"
     ]
     Retries = 5
-
   [API.GraphQL]
     Hosts = [
       "https://api-validators-testnet.vega.rocks/graphql",
-      "https://vega-testnet.anyvalid.com/query"
+      "https://vega-testnet.anyvalid.com/query",
+      "https://testnet.vega.xprv.io/datanode/query"
     ]
-
   [API.REST]
     Hosts = [
       "https://api-validators-testnet.vega.rocks",
-      "https://vega-testnet.anyvalid.com:3009"
+      "https://vega-testnet.anyvalid.com:3009",
+      "https://testnet.vega.xprv.io/datanode"

```

Comment:
When I try: `https://testnet.vega.xprv.io/datanode/api/v2/epoch`, I get a `404 Not Found` response.
Can you re-check your configuration?
I am comparing it with: `https://vega-testnet.anyvalid.com:3009/api/v2/epoch`

Reply:
It appears some of the endpoints are working but some others don't. To continue...

Reply:
Similar to above, working now